src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs

```diff
@@ -125,6 +128,7 @@ private DependencyContext LoadAssemblyContext(Assembly assembly, IDependencyCont
             return null;
         }
 
+        [RequiresAssemblyFiles(Message = "Needs additional support for reading .deps.json in single file", Url = "https://github.com/dotnet/runtime/issues/41265")]
```

Comment:
I would not use the `Url` property for pointing to active issues...
Typically what we do in other places we add a const string which says that the feature is not supported with single-file and what workaround there is (if any) and then use that string in all the attributes which were added because of that feature.

Implementation details can be added in comments if needed.